lib/permissions.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const ADMIN = 'ADMIN';
+const REVIVE_FEATURE = 'REVIVE_FEATURE';
+const CREATE_FEATURE = 'CREATE_FEATURE';
+const UPDATE_FEATURE = 'UPDATE_FEATURE';
+const DELETE_FEATURE = 'DELETE_FEATURE';
+const CREATE_STRATEGY = 'CREATE_STRATEGY';
+const UPDATE_STRATEGY = 'UPDATE_STRATEGY';
+const DELETE_STRATEGY = 'DELETE_STRATEGY';
+const UPDATE_APPLICATION = 'UPDATE_APPLICATION';
+
+function requirePerms(prms) {
+    return (req, res, next) => {
+        for (const permission of prms) {
+            if (
+                req.user &&
+                req.user.permissions &&
+                (req.user.permissions.indexOf(ADMIN) !== -1 ||
+                    req.user.permissions.indexOf(permission) !== -1)
+            ) {
+                return next();
+            }
+        }
+        return res
+            .status(403)
+            .json({
+                message: 'Missing permissions to perform this action.',
+            })
+            .end();
+    };
+}
+
+module.exports = {
+    requirePerms,
+    ADMIN,
+    REVIVE_FEATURE,
```

Comment:
Do we really need "REVIVE_FEATURE"? Would it make sense to allow a user with `UPDATE_FEATURE` rights to be allowed to revive the toggle?

Reply:
I was just thinking the same thing while implementing the permissions in the frontend.

Reply:
Also, would a VIEW_HISTORY permissions be useful?

Reply:
we can add the `VIEW_*` permissions later if needed.